mne/io/nsx/nsx.py

```diff
@@ -0,0 +1,601 @@
+            chan_info["unit_mul"] = FIFF.FIFF_UNITM_NONE
+            chan_info["unit"] = FIFF.FIFF_UNIT_UNITLESS
+            chan_info["range"] = 1
+        else:
+            chan_info["unit_mul"] = FIFF.FIFF_UNITM_NONE
+            chan_info["unit"] = FIFF.FIFF_UNIT_V
+            chan_info["range"] = _unit_range_dict[ch_units[idx]]
+        chan_info["cal"] = cals[idx]
+        chan_info["coord_frame"] = FIFF.FIFFV_COORD_HEAD
+        chan_info["coil_type"] = FIFF.FIFFV_COIL_EEG
+        chan_info["kind"] = FIFF.FIFFV_EEG_CH
+        # montage can't be stored in NSx so channel locs are unknown:
+        chan_info["loc"] = np.full(12, np.nan)
+        orig_units[ch_name] = ch_unit
+
+        # if the NSx info contained channel type information
+        # set it now
+        ch_type = ch_types[idx]
+        if ch_type is not None and ch_type in CH_TYPE_MAPPING:
+            chan_info["kind"] = CH_TYPE_MAPPING.get(ch_type)
+            if ch_type not in ["EEG", "ECOG", "SEEG", "DBS", "CC"]:
+                chan_info["coil_type"] = FIFF.FIFFV_COIL_NONE
+                pick_mask[idx] = False
+        # if user passes in explicit mapping for eog, misc and stim
+        # channels set them here
+        if ch_name in eog or idx in eog or idx - nchan in eog:
+            chan_info["coil_type"] = FIFF.FIFFV_COIL_NONE
+            chan_info["kind"] = FIFF.FIFFV_EOG_CH
+            pick_mask[idx] = False
+        elif ch_name in misc or idx in misc or idx - nchan in misc:
+            chan_info["coil_type"] = FIFF.FIFFV_COIL_NONE
+            chan_info["kind"] = FIFF.FIFFV_MISC_CH
+            pick_mask[idx] = False
+        elif idx in stim_channel_idxs:
+            chan_info["coil_type"] = FIFF.FIFFV_COIL_NONE
+            chan_info["unit"] = FIFF.FIFF_UNIT_NONE
+            chan_info["kind"] = FIFF.FIFFV_STIM_CH
+            pick_mask[idx] = False
+            chan_info["ch_name"] = ch_name
+            ch_names[idx] = chan_info["ch_name"]
+        elif ch_type not in CH_TYPE_MAPPING:
+            chs_without_types.append(ch_name)
+        chs.append(chan_info)
+
+    # warn if channel type was not inferable
+    if len(chs_without_types):
+        msg = (
+            "Could not determine channel type of the following channels, "
+            f'they will be set as EEG:\n{", ".join(chs_without_types)}'
+        )
+        logger.info(msg)
+
+    sfreq = nsx_info["timestamp_resolution"] / nsx_info["period"]
+    info = _empty_info(sfreq)
+    info["meas_date"] = nsx_info["meas_date"]
+    info["chs"] = chs
+    info["ch_names"] = ch_names
+
+    highpass = nsx_info["highpass"][:128]
+    lowpass = nsx_info["lowpass"][:128]
+    if np.all(highpass == highpass[0]):
+        if highpass[0] == "NaN":
+            # Placeholder for future use. Highpass set in _empty_info.
+            pass
+        else:
+            hp = highpass[0]
+            try:
+                hp = float(hp)
+            except Exception:
+                hp = 0.0
```

Comment:
I see a lot of try/except... is it really necessary? I'm more inclined to just do it and if we find problematic files add more targeted checks, like if they have `'N/A'` check for it directly...